app/controllers/foreman_puppet/api/v2/puppet_base_controller.rb

```diff
@@ -1,10 +1,9 @@
 module ForemanPuppet
   module Api
     module V2
-      class BaseController < ::Api::V2::BaseController
+      class PuppetBaseController < ::Api::V2::BaseController
```

Comment:
Is there any other reason for renaming the controllers, or do you just feel it's a better name?

Reply:
The reason is that the `resource_id` of any controller is generated from the name of the controller. As there already exist a controller named `BaseController` in Foreman core, there is a name clash. It is okay for the controllers that should be overwritten anyways, but not for the ones that should remain in Foreman core.